pkg/network/usm/sharedlibraries/ebpf.go

```diff
@@ -96,39 +96,41 @@ func GetEBPFProgram(cfg *ddebpf.Config) *EbpfProgram {
 	progSingletonOnce.Do(func() {
 		progSingleton = &EbpfProgram{
 			cfg:     cfg,
-			libsets: make(map[Libset]*libsetData),
+			libsets: make(map[Libset]*libsetHandler),
+		}
+
+		// Initialize the libsets to avoid requiring a mutex on the map. Golang maps are thread safe
+		// for reads.
+		for libset := range LibsetToLibSuffixes {
+			progSingleton.libsets[libset] = &libsetHandler{
+				callbacks: make(map[*LibraryCallback]struct{}),
+			}
 		}
 	})
 	progSingleton.refcount.Inc()
 
 	return progSingleton
 }
 
-func (e *EbpfProgram) getLibsetData(libset Libset) *libsetData {
-	if data, ok := e.libsets[libset]; ok {
-		return data
-	}
-
-	data := &libsetData{
-		libset:    libset,
-		callbacks: make(map[*LibraryCallback]struct{}),
-		done:      make(chan struct{}),
-	}
-	e.libsets[libset] = data
-	return data
-}
-
+// enableLibset enables the eBPF program for the given libset. Assumes initMutex is locked
 func (e *EbpfProgram) enableLibset(libset Libset) {
```

Comment:
seems like enableLibset is not being called anymore
Did I miss anything?

Reply:
You're correct, it's not needed. Before it was a more complex function but now it's unnecessary.